### DISCUSSION TITLE
How to work with masks in Sd.NEXT? Is it possible?

### DISCUSSION THREAD

**Discussion by blacklig** [hako-mikan/sd-webui-regional-prompter#193] [General]
Hi, thanks a lot for great plugin. Works very well for areas but I cannot get it working for masks.
I have tried to follow up official guid, but just on the first step it breaks apart as I dont have option "First, make sure you switch to **mask divide mode** next to horizontal / vertical. Otherwise the mask will be ignored and regions will be split by ratios as usual."

Is it possible it is missing in Sd.Next (by Vladmandic)? So I was trying without it thinking that if maybe I am playing with masks and putting masks there it would work but it is not. It is like completly ignoring my masks. 

I am doing what is said in guide.. adding masks first via "create mask area" and then adding it via "draw region". So his part works but then when press generate and having prompts (using one from example as well), image just completly ignores those masks.

Am I doing something wrong? :) Thanks for help

**Comment by Symbiomatrix** [comment]
Oh, sorry, that particular line in the guide is from an older version; now there's the matrix / mask / prompt tabs to select the mode instead.
You didn't by any chance tick base / common and didn't write a clause for them, or forgot to tick active? Any errors showing up?
Other than that, it sounds like you followed the guide correctly. Need to check if everything's still operational on vlad's fork.

**Comment by blacklig** [comment]
Seems to me your great plugin just is not working in vlad with masks.. regions are OK... in auto it kinda works and really is interesting to use, but cannot get it going in vlad although I am trying to do the all same. Anyone here having success with masks in vlad?

And by the way - any chance it could work in comfy? Didnt try comfty yet but it is getting lot of traction lately

**Reply by Symbiomatrix** [reply]
I've pretty much gotten vlad's latest up and running locally (bar XL), so I should be able to get back to regional in a few days and see what's up.
Had the same thought myself, I tried it when XL came out. Comfy has a very minimalistic ui based on atomic customisable components, it seems highly optimised, and I find it to be very elegantly designed in general. It has very few extensions as far as I could tell, doesn't offer the relative comfort of gradio doing the heavy lifting in frontend or other packages being available passively. Even if that part can be done, the atomic structure I suspect would necessitate reworking the whole callback design, assuming there is some alternative (possibly nowhere to plug in the regions because there's no standard open flow to the ui). Lotta research to be done.
EDIT: Oh, and one other significant issue is that loras are only implemented as components, rather than invoked in prompt, so they can't be associated with regions without some workaround.
